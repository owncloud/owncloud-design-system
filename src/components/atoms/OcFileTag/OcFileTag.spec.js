import { shallowMount } from "@vue/test-utils"
import Tag from "./OcFileTag.vue"

describe("OcFileTag", () => {
  it("uses correct component when type is specified", () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        type: "button",
      },
    })

    expect(wrapper.element.tagName.toLowerCase()).toMatch("button")
    expect(wrapper).toMatchSnapshot()
  })

  it("emits a click event", async () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        type: "a",
      },
    })

    await wrapper.trigger("click")
    expect(wrapper.emitted().click).toBeTruthy()
  })
})
