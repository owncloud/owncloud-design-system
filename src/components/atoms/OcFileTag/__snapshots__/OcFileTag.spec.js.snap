// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcFileTag uses correct component when type is specified 1`] = `<button class="oc-file-tag oc-file-tag-m oc-tag-button"></button>`;
